raiden_contracts/contracts/TokenNetwork.sol

```diff
@@ -519,49 +497,55 @@ contract TokenNetwork is Utils {
 
     /// @notice Unlocks all locked off-chain transfers and sends the locked tokens to the
     /// participant. Anyone can call unlock on behalf of a channel participant.
-    /// @param channel_identifier The channel identifier - mapping key used for `channels`.
+    /// @param participant Address of the participant who will receive the unlocked tokens.
     /// @param partner Address of the participant who owes the locked tokens.
     /// @param merkle_tree_leaves The entire merkle tree of pending transfers.
     function unlock(
-        uint256 channel_identifier,
         address participant,
         address partner,
         bytes merkle_tree_leaves
     )
         public
     {
-        // Channel must be settled and channel data deleted
-        require(channels[channel_identifier].state == 0);
+        require(participant != 0x0);
+        require(partner != 0x0);
+        require(merkle_tree_leaves.length > 0);
```

Comment:
For example here you forgot to also add the `participant != partner` check. It's much better to put the logic inside the `getChannelIdentifier()` so you don't have to remember it in each place you want to get the channel Identifier.

Reply:
Yes, doing that. good idea.

Reply:
I could also move all the `require(participant != 0x0);` checks there

Reply:
yeah that's correct! Do you only do those checks at the same places where this `getChannelIdentifier()` gets called? If yes it's perfect. If not you can still include it in the function but I am afraid we may forget the call is in there and end up (in the future) having double checks.

But in general should be fine to include that check too as long as we remember it's in the function.

Reply:
>  Do you only do those checks at the same places where this getChannelIdentifier() gets called?

Yes. And I moved them. Better have them doubled than not have them at all anyway.